docs/how-tos/hello-world-module.md

```diff
@@ -0,0 +1,304 @@
+
+{{% snippet "setup.md" %}}
+
+{{% /expand%}}
+
+## What is a module? What is a modular resource?
+
+A module is a set of files that provides support for one or more {{< glossary_tooltip term_id="component" text="components" >}} or {{< glossary_tooltip term_id="service" text="services" >}} that are not built into `viam-server`.
+The {{< glossary_tooltip term_id="resource" text="resources" >}} supported by a module are called modular resources.
+
+## Write the basic functionality into a test script
+
+The point of creating a module is to add functionality to your machine, so before you do anything else, it is helpful to define the functionality that you will later package into a module.
+
+For the purposes of this guide, you're going to make a module that does two things: It opens an image file from a configured path on your machine, and it prints `Hello, World! The latest random number is __.` with a random number in the blank.
+
+1. Find an image you'd like to display when your program runs.
+   We used [this image of a computer with "hello world" on the screen](https://unsplash.com/photos/a-laptop-computer-sitting-on-top-of-a-wooden-desk-8q6e5hu3Ilc).
+   Save the image to your computer.
+
+1. Create a test script file on your computer and copy the following code in the programming language of your choice into it:
+
+    {{< tabs >}}
+{{% tab name="Python" %}}
+
+```python {class="line-numbers linkable-line-numbers"}
+# test.py opens an image and prints a random number
+from PIL import Image
+import random
+
+# TODO: Replace path with path to where you saved your photo
+photo = Image.open("/Users/jessamyt/Downloads/hello-world.jpg")
+
+photo.show()
+
+number = random.random()
+
+print("Hello, World! The latest random number is ", number, ".")
+```
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+```go {class="line-numbers linkable-line-numbers"}
+// test.go opens an image and prints a random number
+package main
+
+import "fmt"
+import "math/rand"
+
+func main() {
+
+  number := rand.Float64()
+  fmt.Println("Hello, World! The latest random number is ", number, ".")
+
+  // TODO
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+1. Replace the path in the script above with the path to where you saved your photo.
+   Save the file.
+
+1. Run the test script in your terminal:
+
+    {{< tabs >}}
+{{% tab name="Python" %}}
+
+It's best practice to use a virtual environment for running Python scripts.
+You'll also need to install the dependency Pillow in the virtual environment before running the test script.
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+source venv/bin/activate
+pip install Pillow
+python3 test.py
+```
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+go run test.go
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+    The image you saved should open on your screen, and a random number should print to your terminal.
+
+## Choose an API to implement
```

Comment:
I love, love, love that this implements 2 apis. I just yesterday heard feedback about how we keep showing people the pattern one module with one resource and that no one will think to do it differently. I think this breaks that pattern.